sw/device/tests/kmac_endianess_test.c

```diff
@@ -0,0 +1,207 @@
+ *
+ * @return OK or error.
+ */
+status_t test_state_endianess(void) {
+  uint32_t out_big_endian[kmac_test.digest_len];
+  uint32_t out_little_endian[kmac_test.digest_len];
+  uint32_t big_to_little[kmac_test.digest_len];
+
+  // Intialize KMAC hardware.
+  dif_kmac_t kmac;
+  dif_kmac_operation_state_t kmac_operation_state;
+  CHECK_DIF_OK(
+      dif_kmac_init(mmio_region_from_addr(TOP_EARLGREY_KMAC_BASE_ADDR), &kmac));
+
+  LOG_INFO("Testing big endian state.");
+  dif_kmac_config_t config = (dif_kmac_config_t){
+      .entropy_mode = kDifKmacEntropyModeSoftware,
+      .entropy_seed = {0xb153e3fe, 0x09596819, 0x3e85a6e8, 0xb6dcdaba,
+                       0x50dc409c, 0x11e1ebd1},
+      .entropy_fast_process = kDifToggleEnabled,
+      .output_big_endian = true,
+  };
+
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+  CHECK_DIF_OK(dif_kmac_mode_kmac_start(&kmac, &kmac_operation_state,
+                                        kmac_test.mode, kmac_test.digest_len,
+                                        &kmac_test.key, NULL));
+  CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                               kmac_test.message_big_endian,
+                               kmac_test.message_len, NULL));
+  CHECK_DIF_OK(dif_kmac_squeeze(&kmac, &kmac_operation_state, out_big_endian,
+                                kmac_test.digest_len, /*processed=*/NULL,
+                                /*capacity=*/NULL));
+  CHECK_DIF_OK(dif_kmac_end(&kmac, &kmac_operation_state));
+
+  // Convert big to little endian.
+  for (int i = 0; i < kmac_test.digest_len; i++) {
+    big_to_little[i] = big_to_little_endian(out_big_endian[i]);
+  }
+
+  LOG_INFO("Testing little endian state.");
+  config.output_big_endian = false;
+
+  CHECK_DIF_OK(dif_kmac_configure(&kmac, config));
+  CHECK_DIF_OK(dif_kmac_mode_kmac_start(&kmac, &kmac_operation_state,
+                                        kmac_test.mode, kmac_test.digest_len,
+                                        &kmac_test.key, NULL));
+  CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state,
+                               kmac_test.message_big_endian,
+                               kmac_test.message_len, NULL));
+  CHECK_DIF_OK(dif_kmac_squeeze(&kmac, &kmac_operation_state, out_little_endian,
+                                kmac_test.digest_len, /*processed=*/NULL,
+                                /*capacity=*/NULL));
+  CHECK_DIF_OK(dif_kmac_end(&kmac, &kmac_operation_state));
+
+  // Compare the two outputs.
+  CHECK_ARRAYS_EQ(out_little_endian, big_to_little, kmac_test.digest_len);
+
+  return OK_STATUS();
+}
+
+/**
+ * Message endianess test.
+ *
+ * This test runs a KMAC operation twice, in the first run with
+ * message_big_endian true and the input message is in big ending. In the second
+ * run, the message_big_endian config is set to false and the input message is
+ * in little endian. In the end, the output digest should match.
```

Comment:
It probably makes sense to also document which digest endianess you're using here.